`DialogPrimitive` with `Modal` and `Flyout` (non-breaking) - Documentation (#2213)
hashicorp/design-system: website/docs/components/modal/partials/guidelines/guidelines.md

An icon paired with the title can help reinforce the purpose and function of the Modal while also drawing the eye to the header and title area. Icons can be used to communicate the severity and importance of interacting with a Modal and are especially useful in a **warning** or **critical** color Modal.

![Modal header title with icon and dismiss button](/assets/components/dialog-primitives/dialog-primitives-header-icon-and-title.jpg)

#### Without title icon

> ![Modal header title only and dismiss button](/assets/components/dialog-primitives/dialog-primitives-header-title-only.jpg)

Reviewer: ```suggestion
![Modal header title with only dismiss button](/assets/components/dialog-primitives/dialog-primitives-header-title-only.jpg)
```

Author: Leaving the original version, so resolving this now.

!!! Do

> ![Modal footer example with two buttons, primary saying 'Rename cluster' and the secondary saying 'Cancel'](/assets/components/modal/modal-action-content-do.png)

Reviewer: ```suggestion
![Modal footer example with two buttons, a primary 'Rename cluster' button and a secondary 'Cancel' button](/assets/components/modal/modal-action-content-do.png)
```

!!!

!!! Dont

> ![Modal footer example with two buttons, primary saying 'Rename this cluster' and the secondary saying 'Dont rename this cluster'](/assets/components/modal/modal-action-content-dont.png)

Reviewer: ```suggestion
![Modal footer example with two buttons, a primary 'Rename this cluster' button and a secondary 'Dont rename this cluster' button](/assets/components/modal/modal-action-content-dont.png)
```
